app/api/margatsni/v1/users/users.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Margatsni
+  module V1
+    module Users
+      class Users < Margatsni::V1::BaseV1
+        helpers do
+          def represent_user_with_token(user)
+            present user, with: Margatsni::V1::Entities::User
+          end
+        end
+
+        namespace :users do
+          desc 'register a user'
+          params do
+            requires :username, type: String
+            requires :email, type: String, regexp: /\A[^@\s]+@[^@\s]+\z/
+            requires :password, type: String
+          end
+          post :registration do
+            user = User.new(declared(params))
+            user.save!
```

Comment:
What if the user wasn't saved because of any validation issue? (for example, email is not unique) 

You would need to write 'if else' condition to process both success and failure response.